src/v/vlog.h

```diff
@@ -28,9 +27,8 @@
     logger.log(                                                                \
       level,                                                                   \
       "{}:{} - " fmt,                                                          \
-      (const char*)&__FILE__[vlog_internal::log_basename_start<                \
-        vlog_internal::basename_index(__FILE__)>::value],                      \
-      __LINE__,                                                                \
+      vlog_internal::file_basename(),                                          \
+      std::source_location::current().line(),                                  \
```

Comment:
nice. wonder if this has any perf impact. the static template-foo was something i wanted to remove a while ago.

Reply:
Maybe? My initial instinct is no, I believe this should result in the same compiled code, but is replacing the macros with the standard feature.

I was actually thinking it might be nice to wrap the base file name + number into a little struct. Maybe something for another PR